src/Features/Core/Portable/GenerateMember/GenerateVariable/AbstractGenerateVariableService.cs

```diff
@@ -216,10 +216,23 @@ private static void GenerateWriteableField(ArrayBuilder<CodeAction> result, Sema
                 if (containingMethod.Parameters.Length > 0)
                 {
                     var compilation = await document.Project.GetRequiredCompilationAsync(cancellationToken).ConfigureAwait(false);
-                    var lastParameterIsCancellationToken = Equals(containingMethod.Parameters.Last().Type, compilation.CancellationTokenType());
 
-                    if (lastParameterIsCancellationToken)
-                        parameterIndex--;
+                    for (var i = containingMethod.Parameters.Length - 1; i >= 0; i--)
+                    {
+                        var parameter = containingMethod.Parameters[i];
+                        if ((parameter.HasExplicitDefaultValue
+                            || Equals(parameter.Type, compilation.CancellationTokenType())
+                            || parameter.IsParams
+                            || parameter.RefKind is RefKind.Out)
+                            && !(containingMethod.IsExtensionMethod && i == 0))
+                        {
+                            parameterIndex--;
+                        }
```

Comment:
💡 Consider breaking this up into multiple if/else if sections, so you can add a comment for each one.

Reply:
+1, or perhaps as an alternative, separate out the extension method check to after this logic. ie, if its an extension method then there could be a guard clause to ensure the minimum value for `i` is 1.